packages/playground/__tests__/Header.test.tsx

```diff
@@ -58,82 +50,68 @@ describe('Given the Header component is rendered', () => {
       expect(screen.queryByText('Dark Mode')).toBeNull();
     });
   });
+});
 
-  test('When defaults are unchanged', () => {
-    render(
-      <AppContext.Consumer>
-        {([state]) => {
-          return (
-            <>
-              <p>Darkmode is {`${state.darkmode ? 'active' : 'inactive'}`}</p>
-              <p>Minimap is {`${state.minimap ? 'active' : 'inactive'}`}</p>
-              <p>Numbering is {`${state.numbering ? 'active' : 'inactive'}`}</p>
-            </>
-          );
-        }}
-      </AppContext.Consumer>
-    );
-    const darkmodeContextValue = screen.getByText('Darkmode is active');
-    const minimapContextValue = screen.getByText('Minimap is active');
-    const numberingContextValue = screen.getByText('Numbering is active');
-
-    // Then darkmode, numbering and minimap are active
-    expect(darkmodeContextValue).toBeInTheDocument();
-    expect(minimapContextValue).toBeInTheDocument();
-    expect(numberingContextValue).toBeInTheDocument();
-  });
-
-  test('When dark mode is toggled', async () => {
+describe('Given the Header interacts with context', () => {
+  beforeEach(() => {
     render(
       <AppProvider>
+        <Header />
         <AppContext.Consumer>
-          {([state, dispatch]) => {
+          {([state]) => {
             return (
               <>
-                <button
-                  onClick={() => {
-                    dispatch({ type: 'SET_DARKMODE', payload: !state.darkmode });
-                  }}
-                >
-                  Toggle Dark Mode
-                </button>
                 <p>Darkmode is {`${state.darkmode ? 'active' : 'inactive'}`}</p>
```

Comment:
Not sure if this makes sense. You are testing the state changes through a made up UI just for testing. We have the reducer test already and we have the tests right above. I think we could savely drop this one.

Reply:
The goal in this one is to test the Header component, including the OverlayPanel and the InputSwitches inside. From an e2e perspective it makes sense to me, because the implementation of the reducer through the components and correct wiring with the context is tested, which would not be cached by the reducer test.